### PR TITLE
Enable mypy `disallow_untyped_defs`

### PR DESCRIPTION
**Pull request by trim21**
depends on https://github.com/PyGithub/PyGithub/pull/2607

### CODE REVIEW

**Comment by trim21** [comment]
@EnricoMi 
image

File: tests/ReplayData/PullRequestReview.setUp.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ None
 None
```
Comment (EnricoMi) [inline tests/ReplayData/PullRequestReview.setUp.txt:18]: why are these two replay data files changing? We ave no changes in test files and github files only touch typing.

Comment (trim21) [reply tests/ReplayData/PullRequestReview.setUp.txt]: There are some code in these 2 replay data and change when running global file replace.

Reverted.

**Comment by trim21** [comment]
we can also enable this option progresive instead or a huge 133 files PR

```toml
[[tool.mypy.overrides]]
module = ["github.AuthenticatedUser"]
check_untyped_defs = true
disallow_untyped_defs = true
```

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Pull request merged**